service/share/cache_availability.go

```diff
@@ -0,0 +1,56 @@
+package share
+
+import (
+	"context"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/autobatch"
+	"github.com/ipfs/go-datastore/namespace"
+)
+
+var cacheAvailabilityPrefix = datastore.NewKey("sampling_result")
+
+func rootKey(root *Root) datastore.Key {
+	return datastore.NewKey(root.String())
+}
+
+// cacheAvailability wraps a given Availability (whether it's light or full)
+// and stores the results of a successful sampling routine over a given Root's hash
+// to disk.
+type cacheAvailability struct {
+	avail Availability
+
+	ds *autobatch.Datastore
+}
+
+// NewCacheAvailability wraps the given Availability with an additional datastore
+// for sampling result caching.
+func NewCacheAvailability(avail Availability, ds datastore.Batching) Availability {
+	ds = namespace.Wrap(ds, cacheAvailabilityPrefix)
+	autoDS := autobatch.NewAutoBatching(ds, 50)
```

Comment:
Make it a global var so that we don't miss this, once we start making all our global vars configurable.

Reply:
https://github.com/celestiaorg/celestia-node/pull/786/commits/0c1f194bb4df5ec64e79341b37715c5fa18f1dba